Tests/Card Tests/CardComponentEventTests.swift

```diff
@@ -0,0 +1,172 @@
+
+        let authPasswordItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.cardViewController.view.findView(with: "AdyenCard.CardComponent.additionalAuthPasswordItem"))
+
+        testFocusEvents(for: authPasswordItemView,
+                        target: .authPassWord,
+                        analyticsProviderMock: analyticsProviderMock)
+    }
+    
+    func testSocialSecurityFocusEvents() throws {
+        let analyticsProviderMock = AnalyticsProviderMock()
+        var config = CardComponent.Configuration()
+        config.socialSecurityNumberMode = .show
+        let sut = makeSUT(with: config, analyticsProviderMock: analyticsProviderMock)
+        
+        let socialSecurityItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.cardViewController.view.findView(with: "AdyenCard.CardComponent.socialSecurityNumberItem"))
+        
+        testFocusEvents(for: socialSecurityItemView,
+                        target: .boletoSocialSecurityNumber,
+                        analyticsProviderMock: analyticsProviderMock)
+    }
+    
+    func testPostalCodeFocusEvents() throws {
+        let analyticsProviderMock = AnalyticsProviderMock()
+        var config = CardComponent.Configuration()
+        config.billingAddress.mode = .postalCode
+        let sut = makeSUT(with: config, analyticsProviderMock: analyticsProviderMock)
+        
+        let postalCodeItemView: FormTextItemView<FormPostalCodeItem> = try XCTUnwrap(sut.cardViewController.view.findView(with: "AdyenCard.CardComponent.postalCodeItem"))
+        
+        testFocusEvents(for: postalCodeItemView,
+                        target: .addressPostalCode,
+                        analyticsProviderMock: analyticsProviderMock)
+    }
+    
+    private func testFocusEvents(
+        for field: FormTextItemView<some FormTextItem>,
+        target: AnalyticsEventTarget,
+        analyticsProviderMock: AnalyticsProviderMock
+    ) {
+        analyticsProviderMock.clearAll()
+        
+        field.textFieldDidBeginEditing(field.textField)
+        field.textFieldDidEndEditing(field.textField)
+        
+        let firstInfoEvent = analyticsProviderMock.infos[0]
+        let secondInfoEvent = analyticsProviderMock.infos[1]
+        
+        XCTAssertEqual(firstInfoEvent.type, .focus)
+        XCTAssertEqual(firstInfoEvent.target, target)
+        
+        XCTAssertEqual(secondInfoEvent.type, .unfocus)
+        XCTAssertEqual(secondInfoEvent.target, target)
+    }
+    
+    private func makeSUT(with configuration: CardComponent.Configuration = .init(), analyticsProviderMock: AnalyticsProviderMock) -> CardComponent {
+        let context = Dummy.context(with: analyticsProviderMock)
+        let cardComponent = CardComponent(paymentMethod: method,
+                                          context: context,
+                                          configuration: configuration)
+        
+        setupRootViewController(cardComponent.viewController)
```

Comment:
The more I read tests the more I feel we can just use 
``` 
let vc = cardComponent.viewController
vs.loadViewIfNeeded
```
This is how we can prevent this test to be an integration test and convert it to a pure unit test that can be executed in a target that has no app launched.